files.js

```diff
@@ -66,12 +36,17 @@ var app_build = [
     "./assets/js/app/Panel.js"
 ];
 
-// Vendor libraries. These are *only* libraries necessary for implementing the plugin.
-// Any vendor libraries needed solely for testing should not appear in this list.
-var vendor_build = [
-    "./assets/js/vendor/d3.min.js",
-    "./assets/js/vendor/q.min.js"
+// App, vendor, and helper files to be included at the top of each test suite
+// NOTE: Order of inclusion is important!
+var test_include = [
+    "./node_modules/should/should.js",
+    ...app_build,
```

Comment:
FYI this is the first appearance of ES6 in the repo and this changes the node dependency for development. I don't use node much so I was still on 4.2.6 (which is suitable for LocusZoom dev) but spread operator support wasn't added until 5.12.0.

Unfortunately upgrading node to current stable (9.2.1) broke other dependencies. Down(up?)grading to 6.4.0, the release that jumped from [59% to 95% ES6 support](http://node.green/) broke similar dependencies (sass-related) but in a slightly different way. Ultimately 5.12.0 worked.

So what I've learned:

* LocusZoom.js's development setup has historically targeted node **4.x**
* With this branch a **minimum node version of 5.12.0 is now required**
* Node **6.4.0 through current stable does not work** for our current development setup
* Old dev dependencies are a root cause of poor node support

I'd be curious to know what you're running locally, seems like we need to thread a needle on this. Those other packages are in big need of being updated.

Also, another dev note: I needed to manually delete existing built files (`locuszoom.app.min.js` in the root dir). Without that they got picked up and mapped by the new build process, barfing out a glut of lint errors.

Reply:
> I'd be curious to know what you're running locally, seems like we need to thread a needle on this. Those other packages are in big need of being updated.

Indeed! Coordinating which node version to use is always a bit of a headache.

I use a program called `nvm` to help me switch between different node versions (hopping between old and new projects is one reason I try to avoid global packages- there's no one answer that fits everything). The file `.nvmrc` gives a clue: I'm using node 8 LTS (carbon). I'm a little surprised that it didn't work for you- are you still having the problem after blowing away `node_modules` and reinstalling? (this PR does include dependency updates)

This also explains the change to the `engines` field of package.json: a warning to others to upgrade. A rough rule of thumb with node is to use odd-numbered versions; 4 is the oldest LTS, and 8 the newest. I'

The node version I use is captured in `.nvmrc`- so if you use

Reply:
> Also, another dev note: I needed to manually delete existing built files (locuszoom.app.min.js in the root dir). Without that they got picked up and mapped by the new build process, barfing out a glut of lint errors.

That's a fair point. My understanding is that few people develop directly on the code- is it fair to assume that we can tell them individually?

We previously git-ignored those files on `develop`, but not on `gh-pages`. 

In addition to being inconsistent, this meant that developers couldn't take easily take advantage of a semi-secret feature that lets NPM install a git repo from a specific commit hash (for people who wanted to drop the newest build into their project while still using package managers to track things exactly).

Reply:
Fixed typos in the above notes; they make more sense read on Github than from an email. Sorry about that.

The most common dependency issue will be sass, but the error should include a command to rebuild and fix the problem.